src/pipecat/services/tavus.py

```diff
@@ -0,0 +1,138 @@
+#
+# Copyright (c) 2024, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+"""This module implements Tavus as a sink transport layer"""
+
+# import asyncio
+# from typing import Any
+import base64
+import requests
+import time
+
+# from enum import Enum
+# from typing import AsyncGenerator
+
+# import numpy as np
+
+from pipecat.frames.frames import ErrorFrame, Frame, TranscriptionFrame, TTSAudioRawFrame, TransportMessageUrgentFrame, TTSStartedFrame, TTSStoppedFrame
+from pipecat.processors.frame_processor import FrameDirection, FrameProcessor
+# from pipecat.services.ai_services import AIService
+from pipecat.transports.base_output import BaseOutputTransport
+from pipecat.transports.services.daily import DailyTransportClient, DailyParams
+# from pipecat.utils.time import time_now_iso8601
+
+from loguru import logger
+
+MIN_AUDIO_BUFFER_SIZE = int(16000 * 2 * 0.5) # 0.5 seconds
+
+class TavusVideoService(BaseOutputTransport):
+    """Class to send base64 encoded audio to Tavus"""
+
+    def __init__(
+        self,
+        *,
+        conversation_id: str,
+        client: DailyTransportClient,
+        params: DailyParams,
+        **kwargs,
+    ) -> None:
+        super().__init__(params, **kwargs)
+        self._client = client
+        self._conversation_id = conversation_id
+        self._tavus_audio_buffer = b""
+
+
+    def can_generate_metrics(self) -> bool:
+        return True
+
+    @classmethod
+    def _get_persona_name(
+        cls,
+        api_key: str,
+        persona_id: str,
+    ) -> str:
+        url = f"https://tavusapi.com/v2/personas/{persona_id}"
+        headers = {
+            "Content-Type": "application/json",
+            "x-api-key": api_key
+        }
+
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+
+        response_json = response.json()
+        logger.debug(f"TavusVideoService persona grabbed {response_json}")
+        return response_json["persona_name"]
+
+    @classmethod
+    def _initiate_conversation(
+        cls,
+        api_key: str,
+        replica_id: str,
+        persona_id: str = "pipecat0",
+        custom_greeting: str = {}
+    ) -> tuple[str, str]:
+        url = "https://tavusapi.com/v2/conversations"
+        headers = {
+            "Content-Type": "application/json",
+            "x-api-key": api_key
+        }
+        payload = {
+            "replica_id": replica_id,
+            "persona_id": persona_id,
+            "custom_greeting": custom_greeting,
+        }
+
+        response = requests.post(url, headers=headers, json=payload)
```

Comment:
`requests` blocks the asyncio event loop. Please, use `aiohttp`.  You can pass an `aiohttp_session` argument as we do in other places. In your example you just have a session and you could just pass it.